### PR TITLE
Fix profiling tests

### PR DESCRIPTION
**Pull request by therault**
Fix the current profiling tests.

Those should also be augmented to include checking the name and values of infos, and a DTD test. This PR only fixes the current tests to adapt to new naming schemes,  the parsec_taskpool_t lifecycle and the new communication engine properties.

### CODE REVIEW

File: tests/profiling/check-comms.py [hunk 1]
```diff
@@ -5,8 +5,8 @@
 t = pd.HDFStore('bw.h5')
 
 result = {
-    'MPI_ACTIVATE': { 'nb': 100, 'lensum': 12000 },
```
Comment (bosilca) [inline tests/profiling/check-comms.py:8]: There is nothing that indicate, even to a seasoned contributor to parsec, that this test need to be in sync with the communication engine. Or that if the test fails one might want to quickly check what is the size of the parsec AM header.

**Comment by bosilca** [comment]
This PR does not fix the test issues, most of the segfault. The checks here pass because we are not testing the profiling. We only run the tests in release mode and the current release mode is compiled without support for profiling because otherwise no PR would succeed.

**Comment by therault** [comment]
> This PR does not fix the test issues, most of the segfault. The checks here pass because we are not testing the profiling. We only run the tests in release mode and the current release mode is compiled without support for profiling because otherwise no PR would succeed.

This is not what I experienced: I did test the PR before pushing (see trace below). Do you have more information how to reproduce the failures?

```
herault@herault-xps:~/Documents/Recherche/parsec/parsec-herault/build/python-tests$ git log -n 1
commit e23b56badd9a5a6f8d44544b316920bf8211d9b0 (HEAD -> fix-profiling-tests, origin/fix-profiling-tests)
Author: Thomas Herault <herault@icl.utk.edu>
Date:   Tue Feb 14 16:31:31 2023 -0500

    Commit 0cb418cc52 added the taskpool name as a prefix to event names
    
    The taskpool must be freed before parsec_fini, or it lingers in a list.
herault@herault-xps:~/Documents/Recherche/parsec/parsec-herault/build/python-tests$ ninja
ninja: no work to do.
herault@herault-xps:~/Documents/Recherche/parsec/parsec-herault/build/python-tests$ ctest -R profiling
Test project /home/herault/Documents/Recherche/parsec/parsec-herault/build/python-tests
    Start 103: profiling/generate_profile_bw:mp
1/7 Test #103: profiling/generate_profile_bw:mp ...   Passed    0.43 sec
    Start 104: profiling/generate_hdf5
2/7 Test #104: profiling/generate_hdf5 ............   Passed    0.72 sec
    Start 105: profiling/check_hdf5
3/7 Test #105: profiling/check_hdf5 ...............   Passed    0.61 sec
    Start 106: profiling/generate_profile_async
4/7 Test #106: profiling/generate_profile_async ...   Passed    0.30 sec
    Start 107: profiling/generate_hdf5_async
5/7 Test #107: profiling/generate_hdf5_async ......   Passed    0.67 sec
    Start 108: profiling/check_hdf5_async
6/7 Test #108: profiling/check_hdf5_async .........   Passed    1.80 sec
    Start 109: profiling/cleanup_profile_files
7/7 Test #109: profiling/cleanup_profile_files ....   Passed    0.00 sec

100% tests passed, 0 tests failed out of 7

Total Test time (real) =   4.55 sec
```

**Comment by bosilca** [comment]
I'm pasting here just one of the failed tests
```
Start testing: Feb 14 16:56 EST
----------------------------------------------------------
18/109 Testing: api/compose
18/109 Test: api/compose
Command: "/Users/bosilca/unstable/parsec/github/build/Debug/tests/api/compose"
Directory: /Users/bosilca/unstable/parsec/github/build/Debug/tests
"api/compose" start time: Feb 14 16:56 EST
Output:
----------------------------------------------------------
[meedy.lan:81234] shmem: mmap: an error occurred while determining whether or not /tmp/ompi.meedy.501/jf.0/2398748672/sm_segment.meedy.501.8efa0000.0 could be created.
[1;37;43mW@00000[0m CPU Features cannot be autodetected on this machine (Detected OSX): No such file or directory
[meedy:81234] *** Process received signal ***
[meedy:81234] Signal: Segmentation fault: 11 (11)
[meedy:81234] Signal code: Invalid permissions (2)
[meedy:81234] Failing at address: 0x0
[meedy:81234] [ 0] 0   libsystem_platform.dylib            0x00000001aba042a4 _sigtramp + 56
[meedy:81234] [ 1] 0   libparsec.4.0.0.dylib               0x0000000102dbdce4 parsec_profiling_trace_flags_info_fn + 672
[meedy:81234] [ 2] 0   libparsec.4.0.0.dylib               0x0000000102dd502c task_profiler_exec_count_end + 264
[meedy:81234] [ 3] 0   libparsec.4.0.0.dylib               0x0000000102dd1764 parsec_pins_instrument + 148
[meedy:81234] [ 4] 0   libparsec.4.0.0.dylib               0x0000000102dab77c __parsec_execute + 1116
[meedy:81234] [ 5] 0   libparsec.4.0.0.dylib               0x0000000102dac2a0 __parsec_task_progress + 208
[meedy:81234] [ 6] 0   libparsec.4.0.0.dylib               0x0000000102dac848 __parsec_context_wait + 992
[meedy:81234] [ 7] 0   libparsec.4.0.0.dylib               0x0000000102dad364 parsec_context_wait + 376
[meedy:81234] [ 8] 0   compose                             0x0000000102b3b960 main + 1764
[meedy:81234] [ 9] 0   dyld                                0x00000001ab6abe50 start + 2544
[meedy:81234] *** End of error message ***
<end of output>
Test time =   0.07 sec
```

**Comment by bosilca** [comment]
And here is a different one
```
101/109 Testing: profiling/check_hdf5
101/109 Test: profiling/check_hdf5
Command: "/opt/local/bin/python3.11" "/Users/bosilca/unstable/parsec/github/parsec/tests/profiling/check-comms.py"
Directory: /Users/bosilca/unstable/parsec/github/build/Debug/tests
"profiling/check_hdf5" start time: Feb 14 16:56 EST
Output:
----------------------------------------------------------
Traceback (most recent call last):
  File "/Users/bosilca/unstable/parsec/github/parsec/tests/profiling/check-comms.py", line 18, in <module>
    evs = t.events[ t.events.type == t.event_types[mt] ]
                    ^^^^^^^^^^^^^
  File "/Users/bosilca/Library/Python/3.11/lib/python/site-packages/pandas/core/generic.py", line 5902, in __getattr__
    return object.__getattribute__(self, name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'DataFrame' object has no attribute 'type'. Did you mean: 'astype'?
Closing remaining open files:bw.h5...done
<end of output>
Test time =   0.35 sec
```

**Comment by therault** [comment]
> 

See commit f01597f51fd1bf597c6ab091593afa3778acaf62 for compose, it's using map_operator.c, and that file was not updated when we added the DSL-specific profiling and task printf functions. This should be fixed.

**Comment by bosilca** [comment]
```
"profiling/check_hdf5" start time: Feb 14 17:35 EST
Output:
----------------------------------------------------------
Traceback (most recent call last):
  File "parsec/tests/profiling/check-comms.py", line 18, in <module>
    evs = t.events[ t.events.type == t.event_types[mt] ]
                    ^^^^^^^^^^^^^
  File "...Python/3.11/lib/python/site-packages/pandas/core/generic.py", line 5902, in __getattr__
    return object.__getattribute__(self, name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'DataFrame' object has no attribute 'type'. Did you mean: 'astype'?
Closing remaining open files:bw.h5...done
```

File: tests/profiling/check-comms.py (hunk 1 above)
Comment (therault) [reply tests/profiling/check-comms.py:8]: See 13cc263d326ab1611db78f17a9e52f0f128399fa

**Comment by therault** [comment]
> And here is a different one
> 
> ```
> 101/109 Testing: profiling/check_hdf5
> 101/109 Test: profiling/check_hdf5
> Command: "/opt/local/bin/python3.11" "/Users/bosilca/unstable/parsec/github/parsec/tests/profiling/check-comms.py"
> Directory: /Users/bosilca/unstable/parsec/github/build/Debug/tests
> "profiling/check_hdf5" start time: Feb 14 16:56 EST
> Output:
> ----------------------------------------------------------
> Traceback (most recent call last):
>   File "/Users/bosilca/unstable/parsec/github/parsec/tests/profiling/check-comms.py", line 18, in <module>
>     evs = t.events[ t.events.type == t.event_types[mt] ]
>                     ^^^^^^^^^^^^^
>   File "/Users/bosilca/Library/Python/3.11/lib/python/site-packages/pandas/core/generic.py", line 5902, in __getattr__
>     return object.__getattribute__(self, name)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> AttributeError: 'DataFrame' object has no attribute 'type'. Did you mean: 'astype'?
> Closing remaining open files:bw.h5...done
> <end of output>
> Test time =   0.35 sec
> ```

That one, I'm unable to reproduce. I tried on my machine and on a fresh pull on Saturn (after digging which python module in saturn has cython... It's python/3.5.2). On these systems, the profiling tests pass (all tests pass).

I can't fix if I can't see it... Can you try running the tests by hand? On saturn that would be

/sw/spack/2022-02-10/opt/spack/linux-scientific7-x86_64/gcc-7.3.0/openmpi-4.1.2-p4eqeh3qsogn7fxjbvm36dtr7paeekz2/bin/mpiexec "-n" "2" "apps/pingpong/bw_test" "-n" "10" "-f" "10" "-l" "2097152" "--" "--mca" "profile_filename" "bw" "--mca" "mca_pins" "task_profiler"
PYTHONPATH=/home/herault/dague/parsec-herault/build/debug/tools/profiling/python/python.test/lib/python3.5/site-packages/:/sw/spack/2022-02-10/opt/spack/linux-scientific7-x86_64/gcc-7.3.0/ninja-1.10.2-jxc6xa3y5cbimcb5a4aaqwrqxrgoogvx/misc /sw/python/3.5.2/bin/python3.5 "/home/herault/dague/parsec-herault/build/debug/tools/profiling/python/profile2h5.py" "--output=bw.h5" "bw-0.prof" "bw-1.prof"

(of course, you need to adapt the PYTHONPATH and program path for your setup... And ctest/cmake removes intermediary files, so you need to generate the prof files from the start)

**Comment by therault** [comment]
> ```
> "profiling/check_hdf5" start time: Feb 14 17:35 EST
> Output:
> ----------------------------------------------------------
> Traceback (most recent call last):
>   File "parsec/tests/profiling/check-comms.py", line 18, in <module>
>     evs = t.events[ t.events.type == t.event_types[mt] ]
>                     ^^^^^^^^^^^^^
>   File "...Python/3.11/lib/python/site-packages/pandas/core/generic.py", line 5902, in __getattr__
>     return object.__getattribute__(self, name)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> AttributeError: 'DataFrame' object has no attribute 'type'. Did you mean: 'astype'?
> Closing remaining open files:bw.h5...done
> ```

See comment about check_hdf5: I don't get these errors and I need more input to find the issue

**Comment by bosilca** [comment]
The problem is that is the hdf5 are not correctly generated (or removed) the tests either report irrelevant errors or report nothing but generate their output file as an empty hdf5. From that point on no test will succeed because all the events and event_types will not be found in the hdf5 file. I pushed a commit that addresses this issue.

**Review by bosilca — APPROVED** [review]

**Pull request merged**